tests/subsys/llext/simple/src/init_preinit_ext.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This test checks for proper support of ELF init arrays. This processing is
+ * performed by llext_bootstrap(), which gets the array of function pointers
+ * from LLEXT via the llext_get_init_array() syscall.
+ *
+ * The proper init sequence would set number to 0xa0a * 0x10 | 0x505 = 0xa5a5.
+ * A different result means that not all initialization routines were executed,
+ * or their order was not correct.
+ */
+
+#include <stdint.h>
+#include <zephyr/llext/symbol.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/ztest_assert.h>
+
+static uint32_t number;
+
+static void preinit_fn_1(void)
+{
+	number = 0xa0a;
+}
+
+static void preinit_fn_2(void)
+{
+	number *= 0x10;
+}
+
+static void init_fn(void)
+{
+	number |= 0x505;
+}
+
+static const void *const preinit_fn_ptrs[] __used Z_GENERIC_SECTION(".preinit_array") = {
+	preinit_fn_1,
+	preinit_fn_2
+};
+static const void *const init_fn_ptrs[] __used Z_GENERIC_SECTION(".init_array") = {
+	init_fn
+};
```

Comment:
can we actually use C++ tests? Might be even nicer to actually have a couple of constructors as a test.